### DISCUSSION TITLE
Debug using Print Statement !!!

### DISCUSSION THREAD

**Discussion by ppandit95** [peridigm/peridigm#217] [General]
Daer Peridigm Community

As I have implemented a new material model for hardening but upon debugging with simple print statements,these statements do not appear while execution of the binary file after making the project using make command . So I wish to know whether I can debug the code in a better way or is there any optimization mode which right away excludes the print statements (which doesn't seems to be the case) .Any help in this regard will prove beneficial

Thanks in Advance
Pushkar

**Comment by jritta** [comment]
Dear Pushkar,

Which print statement do you use? I always used the standard C++ way `std::cout << "Debug print message" << std::endl;`. This works even with MPI for me and it is also used quite often in the code.

Best,
Jonas

**Reply by ppandit95** [reply]
Dear @jritta 

First of all thanks for looking into this query and I am also using the stdard print statement which you have illustrated although I haven't completely able to rectify my code but I did manage to rectify partially using GDB with "-g" flag in cmake

Regards
Pushkar
